src/deepsparse/loggers/helpers.py

```diff
@@ -50,8 +50,52 @@ def possibly_extract_value(value: Any, remainder: Optional[str] = None) -> Any:
     if not remainder:
         return value
 
-    for sub_remainders in remainder.split("."):
-        value = value.__getattribute__(sub_remainders)
+    # splits remainder into separate strings.
+    # each string can access the new nesting depth and optionally
+    # do indexing and slicing at this depth
+    for sub_remainder in remainder.split("."):
+        square_brackets = re.search(r"\[(.*?)\]", sub_remainder)
+        if square_brackets:
+            # retrieve the string without the square brackets
+            sub_remainder = sub_remainder.split("[")[0]
+        # access the new nesting depth
+        value = value.__getattribute__(sub_remainder)
+
+    if square_brackets:
+        return do_slicing_and_indexing(
+            value=value, square_brackets=square_brackets.group(1)
+        )
+
+    return value
+
+
+def do_slicing_and_indexing(
+    value: Union[Sequence, "numpy.ndarray", "torch.tensor"],
+    square_brackets: str,  # noqa F821
+) -> Any:
+    """
+    Perform slicing and/or indexing on the provided value
+
+    Supported operations:
+    - indexing: e.g value[0] or value[-2]
+    - slicing: e.g value[0:2] or value[1:-3]
+    - a composition of both: e.g value[0:2, 0] or value[1:-3, -1]
+
+    :param value: A sequential/tensor/array type variable to be indexed and/or sliced
+    :param square_brackets: The string that contains the indexing and/or slicing
+        information inside square brackets
+    :return: The value of interest
+    """
+    for string_operator in square_brackets.split(","):
+        if ":" in string_operator:
+            # slicing
+            i, j = re.findall(r"-?\d+", string_operator)
+            i, j = int(i), int(j)
```

Comment:
this will fail if they aren't ints, i.e. it's a dictionary lookup with a `:` as a key, should do a check for a dict at the top and just return a get item if so, potentially try checking for numeric keys first